src/mesh.cpp

```diff
@@ -1667,18 +1772,21 @@ UnstructuredMesh::get_bin_from_ent_handle(moab::EntityHandle eh) const {
 }
 
 moab::EntityHandle
-UnstructuredMesh::get_ent_handle_from_bin(int bin) const {
+MOABMesh::get_ent_handle_from_bin(int bin) const
+{
   if (bin >= n_bins()) {
     fatal_error(fmt::format("Invalid bin index: ", bin));
   }
-  return ehs_[bin];
+  return ehs_[0] + bin;
```

Comment:
That is a performance optimization to avoid a search in the `MOAB::Range` data structure. Because we've just loaded the file and haven't modified the mesh, the set of EntityHandles in the MOAB instance is guaranteed to be contiguous, so we can rely on arithmetic here instead of indexing the `Range`.